sway/input/seat.c

```diff
@@ -1462,6 +1462,18 @@ struct sway_node *seat_get_focus(struct sway_seat *seat) {
 	return current->node;
 }
 
+struct wlr_surface *seat_get_focused_surface(struct sway_seat *seat) {
+	struct sway_node *focus = seat_get_focus(seat);
+	if (focus && node_is_view(focus)) {
+		return focus->sway_container->view->surface;
+	}
+	struct sway_seat_node *current = wl_container_of(seat->focus_stack.next, current, link);
+	struct wlr_layer_surface_v1 *layer = current->seat->focused_layer;
```

Comment:
Instead of going through `focus_stack`, can't we just read `seat->focused_layer` directly without the intermediate `current` variable?

Reply:
yes that does indeed work, I copied the code from seat_get_focus without realizing I don't need the node